core/src/test/java/org/elasticsearch/index/query/RegexpQueryBuilderTest.java

```diff
@@ -0,0 +1,146 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.RegexpQuery;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
+import org.joda.time.DateTime;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+
+public class RegexpQueryBuilderTest extends BaseQueryTestCase<RegexpQueryBuilder> {
+
+    private static final String[] REWRITE_METHOD = new String[]{"constant_score_auto", "scoring_boolean", "constant_score_boolean", 
+            "constant_score_filter", "top_terms_boost_1", "top_terms_1"};
+    
+    @Override
+    protected RegexpQueryBuilder doCreateTestQueryBuilder() {
+        RegexpQueryBuilder query = newRandomQueryBuilder();
+
+        if (randomBoolean()) {
+            List<RegexpFlag> flags = new ArrayList<>();
+            for (int i = 0; i < randomInt(5); i++) {
+                flags.add(randomFrom(RegexpFlag.values()));    
+            }
+            query.flags(flags.toArray(new RegexpFlag[flags.size()]));
+        }
+        if (randomBoolean()) {
+            query.maxDeterminizedStates(randomInt(50000));
+        }
+        if (randomBoolean()) {
+            query.rewrite(randomFrom(REWRITE_METHOD));
+        }
+
+        return query;
+    }
+    
+    private RegexpQueryBuilder newRandomQueryBuilder() {
+        String fieldName = null;
+        Object value;
+        switch (randomIntBetween(0, 3)) {
+            case 0:
+                if (randomBoolean()) {
+                    fieldName = STRING_FIELD_NAME;
+                }
+                if (frequently()) {
+                    value = randomAsciiOfLengthBetween(1, 10);
+                } else {
+                    // generate unicode string in 10% of cases
+                    value = randomUnicodeOfLength(10);
+                }
+                break;
+            case 1:
+                if (randomBoolean()) {
+                    fieldName = INT_FIELD_NAME;
+                }
+                value = randomInt(10000);
+                break;
+            case 2:
+                if (randomBoolean()) {
+                    fieldName = DOUBLE_FIELD_NAME;
+                }
+                value = randomDouble();
+                break;
+            case 3:
+                // NO COMMIT: although this is correctly accepted as a date
+                // the regex query will miserably fail with a end-of-string expected at position ..
+                // does it even make sense to have regex on dates?
+                fieldName = DATE_FIELD_NAME_REGEX;
```

Comment:
I am not sure either, I was just wondering what we support and what we do not support. can you test against master?